lib/data/datasource_impl/media/media_data_source_impl.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:fluffychat/data/datasource/media/media_data_source.dart';
+import 'package:fluffychat/data/extensions/media/url_preview_response_extension.dart';
+import 'package:fluffychat/data/network/media/media_api.dart';
+import 'package:fluffychat/domain/model/media/url_preview.dart';
+
+class MediaDataSourceImpl implements MediaDataSource {
+  final MediaAPI _mediaAPI;
+
+  MediaDataSourceImpl(this._mediaAPI);
+
+  @override
+  Future<UrlPreview> getUrlPreview({
+    required Uri uri,
+    int? preferredPreviewTime,
+  }) {
+    return Future.sync(() async {
+      final response = await _mediaAPI.getUrlPreview(
+        uri: uri,
+        preferredPreviewTime: preferredPreviewTime,
+      );
+      return response.toUrlPreview();
+    });
+  }
+}
```

Comment:
what is the difference with that ? 
```suggestion
}  @override
  Future<UrlPreview> getUrlPreview({
    required Uri uri,
    int? preferredPreviewTime,
  }) async {
    final response = await _mediaAPI.getUrlPreview(
      uri: uri,
      preferredPreviewTime: preferredPreviewTime,
    );
    return response.toUrlPreview();
  }
}

```

Reply:
Future.sync is typically used when you have synchronous code (i.e., code that doesn't involve asynchronous operations like network requests or file I/O) that you want to wrap in a Future context. This can be useful in scenarios where you want to ensure that a synchronous operation behaves consistently with asynchronous operations or when you want to use asynchronous control flow constructs with synchronous code.

Reply:
I agree with @imGok 
no need Future.sync here, it just same with Future.value
_mediaAPI.getUrlPreview already is  a Future
![Screenshot 2023-09-19 at 14 11 35](https://github.com/linagora/twake-on-matrix/assets/12546908/da170429-031f-40c5-be7f-edc765524fbf)

Reply:
2 point, i agree with guys @drminh2807 @imGok 
I'll update it